SwiftCheck/Test.swift

```diff
@@ -6,6 +6,78 @@
+///
+///     forAll { (anInteger : Int, aBoolean : Bool, someStrings : ArrayOf<String>) in
+///         return true // This test case always passes.
+///     }
+///
+/// Why require types?  For one, Swift cannot infer the types of local variables because SwiftCheck
+/// uses highly polymorphic testing primitives.  But, more importantly, types are required because 
+/// SwiftCheck uses them to select the appropriate `Gen`erators and shrinkers for each data type
+/// automagically by default.  Those `Gen`erators and shrinkers are then used to create 100 random
+/// test cases that are evaluated lazily to produce a final result.
+///
+/// As mentioned before, SwiftCheck types do not exist in a bubble.  They are highly compositional
+/// and flexible enough to express unique combinations and permutations of test types.  Below is a
+/// purely illustrative example utilizing a significant portion of SwiftCheck's testing functions:
+///
+///     /// This method comes out of SwiftCheck's test suite.
+///     property("Shrunken sets of integers don't always contain [] or [0]") <- forAll { (s : SetOf<Int>) in
+///
+///         /// This part of the property uses `==>`, or the "implication" combinator.  Implication
+///         /// only executes the following block if the preceding statement returns true.  It can
+///         /// be used to discard test cases that contain data you don't want to test with.
+///         return (!s.getSet.isEmpty && s.getSet != Set([0])) ==> {
+///
+///             /// shrinkArbitrary is a method call that invokes the shrinker
+///             let ls = self.shrinkArbitrary(s).map { $0.getSet }
+///             return forAll { (x : Int) in /// OMG, is that a `forAll` inside a `forAll`?!
+///
+///                        return x != x // Well that can't possibly hold.
+///
+///                    }
+///                    ^||^ /// <- `^||^` is like `||` in Swift, but lifted to work with properties.
+///                    (ls.filter({ $0 == [0] || $0 == [] }).count >= 1).withCallback(.AfterTest(.NotCounterexample, { st, res in
+///                        /// For the seriously EXTREME tester, `withCallback` provides a look into
+///                        /// SwiftCheck's testing mechanism.  You can request a variety of information
+///                        /// about the current state of the testing loop from inside this block.
+///                        ///
+///                        /// For the less crazed, SwiftCheck offers straight callbacks with `whenFail` and
+///                        /// `whenEachFail`.
+///                        print("This test is called: \(name)")
+///                    }))
+///         }
+///     }.expectFailure.verbose
+///       ^             ^
+///       |             |
+///       |             +--- The property will print EVERY generated test case to the console.
+///       + --- We expect this property not to hold.
+///
```

Comment:
I've definitely skimmed this, registering it as "interesting, but probably too complicated." The reader needs to know too much about SwiftCheck to understand what this is testing. And then it throws lots of features in all at once. This is the first "real" example, and it feels overwhelming.

It also uses the term "lifted" which many readers won't immediately understand. All you need to say is "`^||^` is like `||` but for properties." Let it be magic for the moment; then remove the magic later by explaining "lift". Sometimes it's ok to explain it parenthetically, by saying something like "technically, this is "lifted." You can read more about lifting \[here\]." But in intro documents, it's best to avoid too many new terms that aren't critical.